azurecontainerapps/src/ContainerAppHelper.ts

```diff
@@ -17,41 +17,314 @@ export class ContainerAppHelper {
+                tl.error(err.message);
+                throw err;
+            }
+    }
+
+    /**
+     * Updates an existing Azure Container App based from a YAML configuration file.
+     * @param containerAppName - the name of the existing Container App
+     * @param resourceGroup - the resource group that the existing Container App is found in
+     * @param yamlConfigPath - the path to the YAML configuration file that the Container App properties will be based from
+     */
+    public updateContainerAppFromYaml(
+        containerAppName: string,
+        resourceGroup: string,
+        yamlConfigPath: string) {
+            tl.debug(`Attempting to update Container App with name "${containerAppName}" in resource group "${resourceGroup}" from provided YAML "${yamlConfigPath}"`);
+            try {
+                let command = `containerapp update -n ${containerAppName} -g ${resourceGroup} --yaml ${yamlConfigPath}`;
+
+                new Utility().throwIfError(
+                    tl.execSync('az', command),
+                    tl.loc('UpdateContainerAppFromYamlFailed')
+                );
+            } catch (err) {
+                tl.error(err.message);
+                throw err;
+            }
+    }
+
+    /**
+     * Determines if the provided Container App exists in the provided resource group.
+     * @param containerAppName - the name of the Container App
+     * @param resourceGroup - the resource group that the Container App is found in
+     * @returns true if the Container App exists, false otherwise
+     */
+    public doesContainerAppExist(containerAppName: string, resourceGroup: string): boolean {
+        tl.debug(`Attempting to determine if Container App with name "${containerAppName}" exists in resource group "${resourceGroup}"`);
+        try {
+            const command = `containerapp show -n ${containerAppName} -g ${resourceGroup} -o none`;
+            const result = tl.execSync('az', command);
+            return result.code == 0;
+        } catch (err) {
+            tl.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided Container App Environment exists in the provided resource group.
+     * @param containerAppEnvironment - the name of the Container App Environment
+     * @param resourceGroup - the resource group that the Container App Environment is found in
+     * @returns true if the Container App Environment exists, false otherwise
+     */
+    public doesContainerAppEnvironmentExist(containerAppEnvironment: string, resourceGroup: string): boolean {
+        tl.debug(`Attempting to determine if Container App Environment with name "${containerAppEnvironment}" exists in resource group "${resourceGroup}"`);
+        try {
+            const command = `containerapp environment show -n ${containerAppEnvironment} -g ${resourceGroup} -o none`;
+            const result = tl.execSync('az', command);
+            return result.code == 0;
```

Comment:
Just double checking if `tl.execSync('az', command)` actually returns non-zero when the app environment does not exist. My initial instinct says an empty response would be returned with a exit code of zero. And return a non-zero code when there's an error in the command itself or the `az` command itself cannot be found.

Reply:
When a non-existent resource is provided to a `show` command (at least the ones I've used for `group`, `containerapp` and `containerapp env`), the command will explicitly throw a `ResourceNotFound` exception, so this returns a non-zero exit code.

Reply:
I double check that a non-zero status code is returned on my end too. I'll defer the review to Dan since he has comments 🙂